src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -36,22 +36,23 @@ static inline int MPIDI_OFI_win_allgather(MPIR_Win * win, void *base, int disp_u
         MPIDI_OFI_WIN(win).mr_key = 0;
     }
 
-    MPIDI_OFI_CALL(fi_mr_reg(MPIDI_Global.domain,       /* In:  Domain Object       */
-                             base,      /* In:  Lower memory address */
-                             win->size, /* In:  Length              */
-                             FI_REMOTE_READ | FI_REMOTE_WRITE,  /* In:  Expose MR for read  */
-                             0ULL,      /* In:  offset(not used)    */
-                             MPIDI_OFI_WIN(win).mr_key, /* In:  requested key       */
-                             0ULL,      /* In:  flags               */
-                             &MPIDI_OFI_WIN(win).mr,    /* Out: memregion object    */
-                             NULL), mr_reg);    /* In:  context             */
-
+    if (MPIDI_OFI_ENABLE_MR_SCALABLE || base) {
```

Comment:
Would be helpful to add a comment here on why we needed this.
You can somehow rephrase the commit message and put it here.

Reply:
Fixed